### PR TITLE
Native Alpha

### PR DESCRIPTION
**Pull request by aapeliv**

### CODE REVIEW

File: app/native/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+   ```bash
+   npm install
+   ```
+
+2. Start the app
+
+   ```bash
+    npx expo start
+   ```
```
Comment (nabramow) [inline app/native/README.md:17]: I think we're using yarn for the rest of the frontend, why npm here?

File: app/native/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- [Expo Go](https://expo.dev/go), a limited sandbox for trying out app development with Expo
+
+You can start developing by editing the files inside the **app** directory. This project uses [file-based routing](https://docs.expo.dev/router/introduction).
+
+## Get a fresh project
+
+When you're ready, run:
+
+```bash
+npm run reset-project
```
Comment (nabramow) [inline app/native/README.md:33]: Same here with yarn

File: app/native/app.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ],
+    "experiments": {
+      "typedRoutes": true
+    },
+    "extra": {
+      "router": {
+        "origin": false
+      },
+      "eas": {
+        "projectId": "ea3d934f-4ba0-4a23-a948-2c259251488f"
```
Comment (nabramow) [inline app/native/app.json:61]: Is this a secret value?

File: app/native/app.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      }
+    },
+    "owner": "couchers-org",
+    "runtimeVersion": {
+      "policy": "appVersion"
+    },
+    "updates": {
+      "url": "https://u.expo.dev/ea3d934f-4ba0-4a23-a948-2c259251488f"
```
Comment (nabramow) [inline app/native/app.json:69]: Same here wondering if it's secret and should be references in an .env var instead

File: app/native/app/(tabs)/couchers.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    void
+  >(() => service.version.status());
+
+  const logIn = () => {
+    authActions.passwordLogin({
+      username: "aapeli_native",
+      password: "unsafepassword",
+      rememberDevice: true,
+    });
+  };
```
Comment (nabramow) [inline app/native/app/(tabs)/couchers.tsx:43]: I mean I know this is probably temporary, but not sure you want a record of the password in the git history?

File: app/native/app/(tabs)/couchers.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+  const logOut = () => {
+    authActions.logout();
+  };
+
+  const onPress = () => {
+    setPressed(pressed + 1);
+    // Alert.alert('Simple Button pressed')
```
Comment (nabramow) [inline app/native/app/(tabs)/couchers.tsx:51]: remove comment

File: app/native/app/(tabs)/couchers.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    newNavState: WebViewNavigation,
+  ) => {
+    const { url } = newNavState;
+    if (!url) return;
+    const v = webview.current;
+    if (!v) return;
+
+    if (!url.startsWith(WEB_BASE_URL)) {
+      v.stopLoading();
+      console.log("oooop");
```
Comment (nabramow) [inline app/native/app/(tabs)/couchers.tsx:65]: You still want the console.log here?

File: app/native/app/(tabs)/couchers.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    }
+  };
+
+  return (
+    <ParallaxScrollView
+      headerBackgroundColor={{ light: "#A1CEDC", dark: "#1D3D47" }}
+      headerImage={
+        <Image
+          source={require("@/assets/images/couchers512.png")}
```
Comment (nabramow) [inline app/native/app/(tabs)/couchers.tsx:74]: Generally it's frowned on to import like this rather than having the imports up top, but I dunno is it a React native thing? Also usually for frontend you do the ES6 import rather than CommonJS imports, i.e. ` import something from "somewhere"`.

File: app/native/app/(tabs)/couchers.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+          onNavigationStateChange={handleWebViewNavigationStateChange}
+          injectedJavaScriptObject={{ isCouchersNativeEmbed: true }}
+        />
+        <ThemedText>(end webview)</ThemedText>
+      </ThemedView>
+      <ThemedView style={styles.stepContainer}>
+        <Button title="Press me" onPress={onPress} />
+        <ThemedText type="subtitle">
+          {pressed == 0 ? "didn't press yet" : `pressed ${pressed} time(s)`}
+        </ThemedText>
```
Comment (nabramow) [inline app/native/app/(tabs)/couchers.tsx:101]: You gonna do the translations later in the `en.json` file or you think these will change a lot?

I just ask as you have some with the i18next references and some not.

File: app/native/app/(tabs)/couchers.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+    height: 350,
+    width: 500,
+    top: 0,
+    right: 0,
+    position: "absolute",
+  },
+  webview: {
+    height: 250,
+  },
+});
```
Comment (nabramow) [inline app/native/app/(tabs)/couchers.tsx:146]: Looks like in other files we mostly put the styles at the top. Just a nit though. I know I'd be looking for them there.

File: app/native/app/(tabs)/explore.tsx [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+    color: "#808080",
+    bottom: -90,
+    left: -35,
+    position: "absolute",
+  },
+  titleContainer: {
+    flexDirection: "row",
+    gap: 8,
+  },
+});
```
Comment (nabramow) [inline app/native/app/(tabs)/explore.tsx:124]: styles up top?

File: app/native/app/+html.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
+import { ScrollViewStyleReset } from "expo-router/html";
+import { type PropsWithChildren } from "react";
+
+/**
+ * This file is web-only and used to configure the root HTML for every web page during static rendering.
+ * The contents of this function only run in Node.js environments and do not have access to the DOM or browser APIs.
+ */
```
Comment (nabramow) [inline app/native/app/+html.tsx:7]: We're not using Node.js though are we since backend is in python?

File: app/native/app/_layout.tsx [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+              />
+              <Drawer.Screen
+                name="account-settings"
+                options={{ drawerLabel: "Account Settings" }}
+              />
+              {/* Divider */}
+              {/* <Drawer.Screen
+                name="help"
+                options={{drawerLabel: 'Help'}}
+              /> */}
```
Comment (nabramow) [inline app/native/app/_layout.tsx:95]: Take these out?

File: app/native/app/everything.tsx [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+    authActions.logout();
+  };
+
+  const onPress = () => {
+    setPressed(pressed + 1);
+    // Alert.alert('Simple Button pressed')
+    checkCoucherCount();
+  };
+
+  const handleWebViewNavigationStateChange = (
```
Comment (nabramow) [inline app/native/app/everything.tsx:55]: This is a duplicate no? What's the reason for the two similar ones?

File: app/native/app/everything.tsx [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+
+    if (!url.startsWith(WEB_BASE_URL)) {
+      v.stopLoading();
+      console.log("oooop");
+    }
+  };
+
+  return (
+    <ParallaxScrollView
+      headerBackgroundColor={{ light: "#A1CEDC", dark: "#1D3D47" }}
```
Comment (nabramow) [inline app/native/app/everything.tsx:71]: I wonder if you want to keep with our color theme and use colors from the existing theme instead? Or add these there?

app/web/theme.ts

File: app/native/assets/images/partial-react-logo.png
Comment (nabramow) [inline app/native/assets/images/partial-react-logo.png:1]: Do you need these?

File: app/native/ci-build-protos.sh [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+elif [[ "$OS" == "Darwin" && "$ARCH" == "arm64" ]]; then
+    OS_ARCH="osx-aarch_64"
+    OS_ARCH_GRPC_WEB="darwin-aarch64"
+else
+    echo "Not one of the known builders!"
+    exit 1
+fi
+
+echo "Using OS/arch: $OS_ARCH"
+
```
Comment (nabramow) [inline app/native/ci-build-protos.sh:27]: I have no idea what this is so just gonna trust you since it's in the separate native folder lol.

File: app/native/components/Alert.tsx [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+    (oldError): oldError is ObscureGrpcErrorMessages =>
+      children.includes(oldError),
+  );
+
+  return (
+    <ThemedText type="error">
+      {
+        // Search for the error in the ugly grpc error object keys
+        // Replace it with the nice error if found
+        oldErrorKey ? grpcErrorStrings[oldErrorKey] : children
```
Comment (nabramow) [inline app/native/components/Alert.tsx:20]: What happens if the `oldErrorKey` is undefined? I think this will throw an error.

File: app/native/components/Collapsible.tsx [hunk 18]
```diff
@@ -0,0 +1,45 @@
+import Ionicons from "@expo/vector-icons/Ionicons";
```
Comment (nabramow) [inline app/native/components/Collapsible.tsx:1]: So we're gonna use Material UI for the frontend and expo for the app, is that right?

File: app/native/components/ExternalLink.tsx [hunk 19]
```diff
@@ -0,0 +1,24 @@
+import { Link } from "expo-router";
+import { openBrowserAsync } from "expo-web-browser";
+import { type ComponentProps } from "react";
```
Comment (nabramow) [inline app/native/components/ExternalLink.tsx:3]: Do you need the type here?

File: app/native/components/ThemedText.tsx [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+    <Text
+      style={[
+        { color },
+        type === "default" ? styles.default : undefined,
+        type === "title" ? styles.title : undefined,
+        type === "defaultSemiBold" ? styles.defaultSemiBold : undefined,
+        type === "subtitle" ? styles.subtitle : undefined,
+        type === "link" ? styles.link : undefined,
+        type === "error" ? styles.error : undefined,
+        type === "button" ? styles.button : undefined,
```
Comment (nabramow) [inline app/native/components/ThemedText.tsx:37]: I don't think you need the undefined here. If the value is not there it will automatically be undefined. So you don't need the ternary operator.

File: app/native/components/ThemedText.tsx [hunk 21]
```diff
@@ -0,0 +1,86 @@
...
+    fontSize: 32,
+    fontWeight: "bold",
+    lineHeight: 32,
+    fontFamily: "Ubuntu_700Bold",
+  },
+  subtitle: {
+    fontSize: 20,
+    fontWeight: "bold",
+    fontFamily: "Ubuntu_700Bold",
+  },
```
Comment (nabramow) [inline app/native/components/ThemedText.tsx:67]: You probably don't want hardcoded pixel fontSizes for mobile since screen sizes differ. Better to use rem. The default of 1rem is 16px, so then fontSize 16px would be 1rem, fontSize 32 would be 2rem, etc.

https://www.sitepoint.com/understanding-and-using-rem-units-in-css/

File: app/native/components/ThemedText.tsx [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+  link: {
+    lineHeight: 30,
+    fontSize: 16,
+    color: "#0a7ea4",
```
Comment (nabramow) [inline app/native/components/ThemedText.tsx:71]: Also eventually probably want to reference these colors via a set theme with primary, secondary, etc, rather than hardcoding.

File: app/native/components/ThemedView.tsx [hunk 23]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+export function ThemedView({
+  style,
+  lightColor,
+  darkColor,
+  ...otherProps
+}: ThemedViewProps) {
+  const backgroundColor = useThemeColor(
+    { light: lightColor, dark: darkColor },
```
Comment (nabramow) [inline app/native/components/ThemedView.tsx:17]: Ooo we love a dark mode :-)

File: app/native/constants/Colors.ts [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  dark: {
+    text: "#ECEDEE",
+    background: "#151718",
+    tint: tintColorDark,
+    icon: "#9BA1A6",
+    tabIconDefault: "#9BA1A6",
+    tabIconSelected: tintColorDark,
+  },
+};
```
Comment (nabramow) [inline app/native/constants/Colors.ts:26]: Oh perfect you did do this, haha. Maybe just reference the others here then!

File: app/native/eas.json [hunk 25]
```diff
@@ -0,0 +1,31 @@
...
+      "distribution": "internal"
+    },
+    "preview": {
+      "distribution": "internal"
+    }
+  },
+  "submit": {
+    "production": {
+      "ios": {
+        "ascAppId": "6623776751"
```
Comment (nabramow) [inline app/native/eas.json:27]: Is this private?

File: app/native/features/auth/useAuthStore.ts [hunk 26]
```diff
@@ -0,0 +1,192 @@
...
+          Sentry.captureException(e, {
+            tags: {
+              component: "auth/useAuthStore",
+              action: "passwordLogin",
+            },
+          });
+          setError(isGrpcError(e) ? e.message : fatalErrorMessage.current);
+        }
+        setLoading(false);
+      },
```
Comment (nabramow) [inline app/native/features/auth/useAuthStore.ts:100]: This is quite hard to read and tell what's going on. Maybe you could break out the functions into separate variables and then just call them within the main function to abstract things out a bit?

File: app/web/.env.development [hunk 27]
```diff
@@ -4,6 +4,4 @@ NEXT_PUBLIC_MEDIA_BASE_URL="https://dev-user-media.couchershq.org"
 NEXT_PUBLIC_CONSOLE_BASE_URL="https://next-console.couchershq.org"
 NEXT_PUBLIC_MAPBOX_KEY="pk.eyJ1IjoiY291Y2hlcnMiLCJhIjoiY2tpYzY4eHd2MGVpcTJ0bGdhdGhvbHhlbiJ9.u5zvDeFE9H8itTK_dNp7Pg"
 NEXT_PUBLIC_NOMINATIM_URL="https://nominatim.openstreetmap.org/"
-NEXT_PUBLIC_TILE_URL="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-NEXT_PUBLIC_TILE_ATTRIBUTION="Map data &copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors"
-NEXT_PUBLIC_STRIPE_KEY="pk_test_51KEzByIfR5z29g5khFE5samD8XKOGLcCrM1lhCkfOomGPUFAEYOw8uAqI2Nkv33wYdPM2FgTQNTC07IiNfHY1kLJ00Jqm8Ppai"
\ No newline at end of file
+NEXT_PUBLIC_STRIPE_KEY="pk_test_51KEzByIfR5z29g5khFE5samD8XKOGLcCrM1lhCkfOomGPUFAEYOw8uAqI2Nkv33wYdPM2FgTQNTC07IiNfHY1kLJ00Jqm8Ppai"
```
Comment (nabramow) [inline app/web/.env.development:7]: Does this have to do with the app? If other cleanup might be better to break into another PR. Then if anything goes wrong easier to revert it.

File: app/web/.env.localdev [hunk 28]
```diff
@@ -5,8 +5,6 @@ NEXT_PUBLIC_CONSOLE_BASE_URL="http://localhost:10027"
 NEXT_PUBLIC_IS_POST_BETA_ENABLED=true
 NEXT_PUBLIC_MAPBOX_KEY="pk.eyJ1IjoiY291Y2hlcnMiLCJhIjoiY2tpYzY4eHd2MGVpcTJ0bGdhdGhvbHhlbiJ9.u5zvDeFE9H8itTK_dNp7Pg"
 NEXT_PUBLIC_NOMINATIM_URL="https://nominatim.openstreetmap.org/"
-NEXT_PUBLIC_TILE_URL="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-NEXT_PUBLIC_TILE_ATTRIBUTION="Map data &copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors"
```
Comment (nabramow) [inline app/web/.env.localdev:9]: Same here, doesn't seem related to this PR?

File: app/web/.env.next [hunk 29]
```diff
@@ -4,6 +4,4 @@ NEXT_PUBLIC_MEDIA_BASE_URL="https://dev-user-media.couchershq.org"
 NEXT_PUBLIC_CONSOLE_BASE_URL="https://next-console.couchershq.org"
 NEXT_PUBLIC_MAPBOX_KEY="pk.eyJ1IjoiY291Y2hlcnMiLCJhIjoiY2tpYzVnaXA0MGdmejJ4bGFrdTdrNjA0NCJ9.1VAPRbE6a_kwGw1kucppJw"
 NEXT_PUBLIC_NOMINATIM_URL="https://nominatim.openstreetmap.org/"
-NEXT_PUBLIC_TILE_URL="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-NEXT_PUBLIC_TILE_ATTRIBUTION="Map data &copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors"
```
Comment (nabramow) [inline app/web/.env.next:8]: Same

File: app/web/.env.production [hunk 30]
```diff
@@ -4,6 +4,4 @@ NEXT_PUBLIC_MEDIA_BASE_URL="https://user-media.couchershq.org"
 NEXT_PUBLIC_CONSOLE_BASE_URL="https://console.couchers.org"
 NEXT_PUBLIC_MAPBOX_KEY="pk.eyJ1IjoiY291Y2hlcnMiLCJhIjoiY2tpYzVnaXA0MGdmejJ4bGFrdTdrNjA0NCJ9.1VAPRbE6a_kwGw1kucppJw"
 NEXT_PUBLIC_NOMINATIM_URL="https://nominatim.openstreetmap.org/"
-NEXT_PUBLIC_TILE_URL="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-NEXT_PUBLIC_TILE_ATTRIBUTION="Map data &copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors"
```
Comment (nabramow) [inline app/web/.env.production:8]: Also same, especially as this PR is so large already, would be great to do this separately.

File: app/web/appConstants.ts [hunk 31]
```diff
@@ -13,8 +13,6 @@ export const pingInterval = 10000;
 
 export const reactQueryRetries = 1;
 
-export const grpcTimeout = 10000; //milliseconds
-
```
Comment (nabramow) [inline app/web/appConstants.ts:17]: Does this have to do with the app?

File: app/web/features/auth/login/Login.tsx [hunk 32]
```diff
@@ -9,15 +9,12 @@ import { useRouter } from "next/router";
 import { useEffect } from "react";
 import vercelLogo from "resources/vercel.svg";
 import { dashboardRoute, signupRoute } from "routes";
-import makeStyles from "utils/makeStyles";
```
Comment (nabramow) [inline app/web/features/auth/login/Login.tsx:12]: Cleanup in a separate PR maybe. I know this is kinda harmless but again it's better to prevent bugs in case something needs to be reverted. Or to track changes that might have caused a bug.

File: app/web/features/auth/login/LoginForm.tsx [hunk 33]
```diff
@@ -1,16 +1,16 @@
...
 import { useAuthContext } from "features/auth/AuthProvider";
 import useAuthStyles from "features/auth/useAuthStyles";
 import { useTranslation } from "i18n";
 import { AUTH, GLOBAL } from "i18n/namespaces";
+import Sentry from "platform/sentry";
 import { useState } from "react";
 import { Controller, useForm } from "react-hook-form";
 import { resetPasswordRoute } from "routes";
-import isGrpcError from "utils/isGrpcError";
+import isGrpcError from "service/utils/isGrpcError";
```
Comment (nabramow) [inline app/web/features/auth/login/LoginForm.tsx:13]: Are these related to the app?

File: app/web/features/auth/useAuthStore.ts [hunk 34]
```diff
@@ -161,12 +135,47 @@ export default function useAuthStore() {
...
   );
 
+  // useEffect(() => {
+  //   // if we aren't logged in and are otherwise idle, but auth state changed, check if the cookie is set in the bg
+  //   if (typeof window !== "undefined" && !authenticated && !loading && !error && !checkedAuthStatus) {
+  //     setCheckedAuthStatus(true);
+  //     authActions.checkAuthStatus();
+  //   }
+  // }, [authenticated, checkedAuthStatus, setCheckedAuthStatus, authActions]);
+
```
Comment (nabramow) [inline app/web/features/auth/useAuthStore.ts:178]: Remove?

File: app/web/platform/nativeLink.ts [hunk 35]
```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+
+export function getReactNativeWebView(): typeof window.ReactNativeWebView {
```
Comment (nabramow) [inline app/web/platform/nativeLink.ts:3]: Does React Native not have specific types exported? Not a big deal just seems more readable and that's how I usually see it.

File: app/web/platform/nativeLink.ts [hunk 36]
```diff
@@ -0,0 +1,57 @@
...
+  if (
+    typeof window !== "undefined" &&
+    window.ReactNativeWebView !== undefined
```
Comment (nabramow) [inline app/web/platform/nativeLink.ts:6]: I think this can just be `typeof window && window.ReactNativeWebView`. You can see it here or test yourself in the dev console.

Screenshot 2024-08-29 at 8 25 33 PM

File: app/web/platform/nativeLink.ts [hunk 37]
```diff
@@ -0,0 +1,57 @@
...
+  ) {
+    return window.ReactNativeWebView;
+  }
+}
+
+export function isNativeEmbed(): boolean {
+  return getReactNativeWebView() !== undefined;
```
Comment (nabramow) [inline app/web/platform/nativeLink.ts:13]: Don't need the undefined unless you just want it to be super explicit.

File: app/web/platform/nativeLink.ts [hunk 38]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+export function getNativeData() {
+  const wv = getReactNativeWebView();
+  if (!wv) return;
+  if (wv.injectedObjectJson()) {
+    return JSON.parse(wv.injectedObjectJson());
+  }
+}
```
Comment (nabramow) [inline app/web/platform/nativeLink.ts:22]: Do y'all have a policy on abbreviations? I usually work places where we try to avoid them for readability but just a nit.

File: app/web/platform/nativeLink.ts [hunk 39]
```diff
@@ -0,0 +1,57 @@
...
+
+export function useIsNativeEmbed(): boolean {
+  const [isNative, setIsNative] = useState(false);
+
+  useEffect(() => {
+    setIsNative(
+      typeof window !== "undefined" && window.ReactNativeWebView !== undefined
```
Comment (nabramow) [inline app/web/platform/nativeLink.ts:29]: Don't need undefined.

File: app/web/platform/nativeLink.ts [hunk 40]
```diff
@@ -0,0 +1,57 @@
...
+  if (!isNativeEmbed()) return;
+  getReactNativeWebView()!.postMessage(
+    JSON.stringify({ type: type, data: data })
+  );
+}
+
+export function getState<T>(key: string) {
+  if (!isNativeEmbed()) return undefined;
+  const data = getNativeData();
+  // if ()
```
Comment (nabramow) [inline app/web/platform/nativeLink.ts:48]: remove?

File: app/web/service/client.ts [hunk 41]
```diff
@@ -93,10 +99,7 @@ const client = {
   threads: new ThreadsPromiseClient(URL, null, opts),
 };
 
-if (
-  process.env.NEXT_PUBLIC_COUCHERS_ENV !== "prod" &&
-  typeof window !== "undefined"
-) {
+if (!IS_PROD && typeof window !== "undefined") {
```
Comment (nabramow) [inline app/web/service/client.ts:102]: I think you can just do window and not typeof window no? Because if it's not undefined itself then it definitely has a type.

File: app/web/service/conversations.ts [hunk 42]
```diff
@@ -19,10 +19,10 @@ import {
...
   SendMessageReq,
 } from "proto/conversations_pb";
-import client from "service/client";
-import { duration2pb } from "utils/date";
-import { Duration } from "utils/dayjs";
-import isGrpcError from "utils/isGrpcError";
+
+import client from "./client";
+import { Duration, duration2pb } from "./utils/date";
+import isGrpcError from "./utils/isGrpcError";
```
Comment (nabramow) [inline app/web/service/conversations.ts:25]: Why change to the specific dot exports? Was there a circular dependency?

File: docs/web-frontend.md [hunk 43]
```diff
@@ -23,8 +23,6 @@ NEXT_PUBLIC_CONSOLE_BASE_URL="http://localhost:10027"
 NEXT_PUBLIC_IS_POST_BETA_ENABLED=true
 NEXT_PUBLIC_MAPBOX_KEY="pk.eyJ1IjoiY291Y2hlcnMiLCJhIjoiY2tpYzY4eHd2MGVpcTJ0bGdhdGhvbHhlbiJ9.u5zvDeFE9H8itTK_dNp7Pg"
 NEXT_PUBLIC_NOMINATIM_URL="https://nominatim.openstreetmap.org/"
-NEXT_PUBLIC_TILE_URL="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
-NEXT_PUBLIC_TILE_ATTRIBUTION="Map data &copy; <a href="https://osm.org/copyright">OpenStreetMap</a> contributors"
```
Comment (nabramow) [inline docs/web-frontend.md:27]: Does this have to do with the app?

**Comment by nabramow** [comment]
Okay I have looked through the code. Still wanna click around the web app to make sure everything looks good. Might not have a chance til the weekend though.

One big picture question though. I see there is a lot of copied in code for React Native. Will it all be used right away? If not, I'd argue to copy it in as you go for three reasons. 1) If anything causes a bug, it's easier to revert that PR for that specific thing, 2) It's way less overwhelming to review, 3) Decreases the chance that we'll end up with unused code we copied and forgot about.


Please take this review with a grain of salt since I am new to the codebase and new to react native haha. Might want one more set of eyes on the web stuff.

File: docs/web-frontend.md (hunk 43 above)
Comment (aapeliv) [reply docs/web-frontend.md:27]: moved to other PR

File: app/web/.env.development (hunk 27 above)
Comment (aapeliv) [reply app/web/.env.development:7]: Moved to #4690

File: app/web/.env.localdev (hunk 28 above)
Comment (aapeliv) [reply app/web/.env.localdev:9]: Moved to #4690

File: app/web/.env.next (hunk 29 above)
Comment (aapeliv) [reply app/web/.env.next:8]: Moved to #4690

File: app/web/.env.production (hunk 30 above)
Comment (aapeliv) [reply app/web/.env.production:8]: Moved to #4690

File: app/web/appConstants.ts (hunk 31 above)
Comment (aapeliv) [reply app/web/appConstants.ts:17]: Moved to #4690

File: app/web/features/auth/login/Login.tsx (hunk 32 above)
Comment (aapeliv) [reply app/web/features/auth/login/Login.tsx:12]: Moved to #4690

File: app/web/features/auth/login/LoginForm.tsx (hunk 33 above)
Comment (aapeliv) [reply app/web/features/auth/login/LoginForm.tsx:13]: Moved to #4690

File: app/web/features/auth/useAuthStore.ts (hunk 34 above)
Comment (aapeliv) [reply app/web/features/auth/useAuthStore.ts:178]: Moved to https://github.com/Couchers-org/couchers/pull/4693

File: app/web/platform/nativeLink.ts (hunk 37 above)
Comment (aapeliv) [reply app/web/platform/nativeLink.ts:13]: fixed

File: app/web/platform/nativeLink.ts (hunk 36 above)
Comment (aapeliv) [reply app/web/platform/nativeLink.ts:6]: fixed

File: app/web/platform/nativeLink.ts (hunk 39 above)
Comment (aapeliv) [reply app/web/platform/nativeLink.ts:29]: fixed

File: app/web/platform/nativeLink.ts (hunk 38 above)
Comment (aapeliv) [reply app/web/platform/nativeLink.ts:22]: fixed

File: app/web/platform/nativeLink.ts (hunk 35 above)
Comment (aapeliv) [reply app/web/platform/nativeLink.ts:3]: I don't think there are TypeScript types for `window.ReactNativeWebView`. I've searched the [repo](https://github.com/react-native-webview/react-native-webview) and couldn't find anything... I am not entirely sure how to do this nicer than what I have here, open to suggestions!

**Comment by aapeliv** [comment]
Superseded by https://github.com/Couchers-org/couchers/pull/4694, https://github.com/Couchers-org/couchers/pull/4690, https://github.com/Couchers-org/couchers/pull/4693, and a to-be PR with the actual native stuff.
